x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/annotations/AnnotationPersister.java

```diff
@@ -58,12 +60,12 @@ public AnnotationPersister(ResultsPersisterService resultsPersisterService) {
      * @param annotation annotation to be persisted
      * @return tuple of the form (annotation id, annotation object)
      */
-    public Tuple<String, Annotation> persistAnnotation(@Nullable String annotationId, Annotation annotation) {
+    public Optional<Tuple<String, Annotation>> persistAnnotation(@Nullable String annotationId, Annotation annotation) {
```

Comment:
I think it's nice to let this throw an exception (as it was) and for the caller to decide what to do with it.

I have the impression that the reason you preferred to change `executeRequest` to catch exceptions was to avoid having to wrap this method with `try-catch` twice in `DatafeedJob`. See there for a suggestion.

Reply:
I suppose this can continue to throw, I am very tempted to make the results persister throw a checked exception...